File: src/chapter4/4-5.ts
let counter = 0;

export const solution = (k: number, currentN: number, useFlag: number) => {
  if (currentN > k) return;
  if (useFlag === 0b111) counter++;

  solution(k, currentN * 10 + 7, useFlag | 0b001);
  solution(k, currentN * 10 + 5, useFlag | 0b010);
  solution(k, currentN * 10 + 3, useFlag | 0b100);
};

File: src/chapter4/20250415/end_of_chapter/4.5.py
counter = 0


def count(K: int, cur: int, use: int):
    if cur > K:
        return 0
    if counter == 0b111:
        counter += 1

    count(K, cur * 10 + 7, use | 0b001)
    count(K, cur * 10 + 5, use | 0b010)
    count(K, cur * 10 + 3, use | 0b100)


def main():
    N = int(input())
    count(N, 0, 0)
    print(counter)
